Add file handler documents (#7711)
GoogleChrome/developer.chrome.com: site/en/docs/extensions/mv3/manifest/file_handlers/index.md

---
layout: "layouts/doc-post.njk"
title: "file_handlers"
seoTitle: "Chrome Extensions Manifest: file_handlers"
date: 2023-11-01
description: Reference documentation for the file_handlers property of manifest.json.
---

> The `"file_handlers"` manifest key specifies file types to be handled by an extension. To process a file, use the web platform's [Launch Handler API](https://developer.mozilla.org/docs/Web/API/Launch_Handler_API). For extension specific information, see [File Handling](/docs/extensions/mv3/file_handling).

Reviewer: ```suggestion
The `"file_handlers"` manifest key specifies file types to be handled by a ChromeOS extension. To process a file, use the web platform's [Launch Handler API](https://developer.mozilla.org/docs/Web/API/Launch_Handler_API). For extension specific information, see [File Handling](/docs/extensions/mv3/file_handling).
```
I'd like to be more explicit that this is only available on ChromeOS

---
layout: "layouts/doc-post.njk"
title: "file_handlers"
seoTitle: "Chrome Extensions Manifest: file_handlers"
date: 2023-11-01

> description: Reference documentation for the externally_connectable property of manifest.json.

Author: ```suggestion
description: Reference documentation for the file_handlers property of manifest.json.
```

---

> 

Reviewer 2: Can we add an aside about availability here? I think identical wording to what I used here would work well: https://developer.chrome.com/docs/extensions/reference/readingList/

Author: I added it below the opening paragraph.

We need a way to automate stuff like this. Such comments have a way of sticking around way past their expiration dates.

The `"file_handlers"` manifest key specifies file types to be handled by an extension. To process a file, use the web platform's [Launch Handler API](https://developer.mozilla.org/docs/Web/API/Launch_Handler_API). For extension specific information, see [File Handling](/docs/extensions/mv3/file_handling).

> 

Author: ```suggestion

{% Aside %}
This feature is currently available in Chrome 120 [beta](https://www.google.com/chrome/beta/).
{% endAside %}

```

```json
"file_handlers": [
  {
    "action": "/open_text.html",
    "name": "Plain text",
    "accept": {
      "text/plain": [".txt"]
    },

>     "icons": [

Reviewer 2: Do we support icons? Genuine question, I'm not certain :)
